### ISSUE TITLE
Default template config does not load (under IISExpress or web)

### ISSUE THREAD

**Issue by zenith-nz** [RehanSaeed/ASP.NET-MVC-Boilerplate#41]
HTTP Error 500.19 - Internal Server Error
The requested page cannot be accessed because the related configuration data for the page is invalid.

Detailed Error Information:
Module	   IIS Web Core
Notification	   Unknown
Handler	   Not yet determined
Error Code	   0x8007000d
Config Error	   
Config File	   \\?\C:\Users\User\Documents\new\test2\WebApplication1\wwwroot\web.config
Requested URL	   http://localhost:15047/
Physical Path	   
Logon Method	   Not yet determined
Logon User	   Not yet determined
Request Tracing Directory	   C:\Users\User\Documents\IISExpress\TraceLogFiles\

Config Source:
   -1: 
    0: 

More Information:
This error occurs when there is a problem reading the configuration file for the Web server or Web application. In some cases, the event logs may contain more information about what caused this error.
If you see the text "There is a duplicate 'system.web.extensions/scripting/scriptResourceHandler' section defined", this error is because you are running a .NET Framework 3.5-based application in .NET Framework 4. If you are running WebMatrix, to resolve this problem, go to the Settings node to set the .NET Framework version to ".NET 2". You can also remove the extra sections from the web.config file.
View more information »

**Comment by RehanSaeed** [comment]
Which versions of .NET, IIS Express, Full IIS, Visual Studio, MVC and Boilerplate are you using?

**Comment by zenith-nz** [comment]
* .NET = for the project in question, dnx452-beta8-x64 runtime under MVC6
* VS = VS2015 Community
* IIS Express = whatever the latest version is in my particular flavour of VS is, as of 11 November 2015 (not at work, so don't know currently) 
* OS = Windows 10 Pro x64

**Comment by zenith-nz** [comment]
Oh, and I tried Boilerplate versions 2.6.0 and 2.7.0. The 2.6 type version I made via the wizard, using defaults provided by your UI option for that. The 2.7 version I made by taking the MVC6 sample part of the source and stripping out the parts that are conditionally included (as per what the user selects in the wizard). 

Both attempts were made in fresh solution files with no other projects present.

**Comment by zenith-nz** [comment]
Oh yes. It's worthy of note that I have a few older projects that were made from your v2.5.0 source. This is how this came about actually -  I was looking to migrate them to the newer template.
They work just fine still. As do the MS-supplied MVC6 templates. 

So I suspect it's not an introduced misconfiguration of my dev machine, or else I'd expect this behaviour to be more widespread, and for it to have affected the older instances. It's possible though, of course.

**Comment by Joe4evr** [comment]
I have the same issue as well. I get the error in the OP when running via IIS, or when using the `dnx web` command and browsing to `localhost:5000`, the browser will just stay infinitely loading the page without getting a response.

**Comment by zenith-nz** [comment]
Discovered what this is. I hadn't installed the Beta 8 Web Tools. I had upgraded dnvm and the DNX runtimes, but not the web tools. @Joe4evr , try installing them from https://www.microsoft.com/en-us/download/details.aspx?id=49442 .

What a cryptic error for this!
Also, it's weird that when I specified the project to use beta7 instead of beta8 it still failed to run (same behaviour). Is this because, somehow, the libraries that this Boilerplate or its dependencies uses beta8, and that despite my specifying using beta7, it tried to use beta8 and subsequently couldn't run properly? I'd have expected a totally different category of error(s) in that case, e.g. type/assembly loading errors, for example.
